pkg/integrations/v4/dm/emitter_test.go

```diff
@@ -249,7 +251,12 @@ func TestEmitter_Send_failedToSubmitMetrics_dropAndLog(t *testing.T) {
 	ms.wg.Wait()
 	ctx.SendDataWg.Wait()
 
-	entry := hook.LastEntry()
+	var entry *logrus.Entry
+	assert.Eventually(t, func() bool {
+		entry = hook.LastEntry()
+		return entry != nil
+	}, time.Second, 10*time.Millisecond)
+
```

Comment:
Maybe we need to add the others assertions inside this func. If not `require.NotNil(t, entry)` could failed if it is not set yet. 🤔

Reply:
Can't add an assertion in the function cos that will fail the test while it was running the "Eventually" loop. The loop will keep running till `entry != nil` so if the loop runs twice the test will fail on the first check if you add an assertion here. The next line we have a `require.NotNil(t, entry)` so that will fail the test straight away if the `assert.Eventually` fails.

Reply:
Ohhh I see, wasn't sure about if it was block until that condition. Thanks for explaining!